File: CCIP/CCIP/InvalidNetworkMessageViewController.m
//
//  InvalidNetworkMessageViewController.m
//  CCIP
//
//  Created by FrankWu on 2016/8/15.
//  Copyright © 2016年 CPRTeam. All rights reserved.
//

#import "InvalidNetworkMessageViewController.h"
#import "AppDelegate.h"
#import "UIColor+addition.h"
#import "UIView+addition.h"
#import "MainNavViewController.h"

@interface InvalidNetworkMessageViewController ()

@property (readwrite, nonatomic) BOOL isRelayout;
@property (strong, nonatomic) NSString *message;

@end

@implementation InvalidNetworkMessageViewController

- (void)setMessage:(NSString *)message {
    _message = message;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    [self.closeButton setTitle:NSLocalizedString(@"InvalidNetworkButtonRetry", nil)
                      forState:UIControlStateNormal];
    [self.closeButton setTintColor:[UIColor whiteColor]];
    [self.closeButton setBackgroundColor:[UIColor colorWithRed:61/255.0 green:152/255.0 blue:60/255.0 alpha:1]];
    [self.closeButton setGradientColor:[AppDelegate AppConfigColor:@"MessageButtonLeftColor"]
                                    To:[AppDelegate AppConfigColor:@"MessageButtonRightColor"]
                            StartPoint:CGPointMake(-.4f, .5f)
                               ToPoint:CGPointMake(1, .5f)];
    CALayer *layer = [self.closeButton.layer.sublayers firstObject];
    [layer setCornerRadius:self.closeButton.frame.size.height / 2];
    [self.closeButton.layer setCornerRadius:self.closeButton.frame.size.height / 2];
    
    [self.messageLabel setText:self.message];
    [self.view setAutoresizingMask:UIViewAutoresizingNone];
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(appplicationDidEnterBackground:)
                                                 name:UIApplicationDidEnterBackgroundNotification
                                               object:nil];
}

- (void)viewWillLayoutSubviews {
    [super viewWillLayoutSubviews];
    if (self.isRelayout != true) {
        MainNavViewController *mnvc = (MainNavViewController *)self.presentingViewController;
        CheckinViewController *cvc = (CheckinViewController *)[[mnvc childViewControllers] firstObject];
        CGFloat topStart = [cvc controllerTopStart];
        topStart = 0;
        self.view.frame = CGRectMake(0.0f,
                                     -44.0f + topStart,
                                     self.view.frame.size.width,
                                     self.view.frame.size.height - topStart - 49.0f + X_TOP(0.0f, 22.0f));
        self.view.superview.frame = CGRectMake(0.0f,
                                               topStart,
                                               self.view.frame.size.width,
                                               self.view.frame.size.height + X_TOP(0.0f, 22.0f));
        self.isRelayout = true;
    }
}

- (IBAction)clossView:(id)sender {
    [self dismissViewControllerAnimated:YES
                             completion:^{
                                 if (self.delegate && [self.delegate respondsToSelector:@selector(refresh)]) {
                                     [self.delegate refresh];
                                 }
                             }];
}

- (void)appplicationDidEnterBackground:(NSNotification *)notification {
    [self dismissViewControllerAnimated:YES
                             completion:nil];
}

- (void)viewDidDisappear:(BOOL)animated {
    [super viewDidDisappear:animated];
    [self dismissViewControllerAnimated:YES
                             completion:nil];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: OPass/Tabs/CheckinTab/CheckinOverlay/InvalidNetworkMessageViewController.swift
//
//  InvalidNetworkMessageViewController.swift
//  OPass
//
//  Created by 腹黒い茶 on 2019/6/17.
//  2019 OPass.
//

import Foundation
import UIKit

class InvalidNetworkMessageViewController: UIViewController {
    public var delegate: InvalidNetworkRetryDelegate?
    public var message: String = ""
    @IBOutlet public var messageLabel: UILabel?
    @IBOutlet public var closeButton: UIButton?

    private var isRelayout: Bool = false

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        self.closeButton?.setTitle(NSLocalizedString("InvalidNetworkButtonRetry", comment: ""), for: .normal)
        self.closeButton?.tintColor = .white
        self.closeButton?.backgroundColor = UIColor.init(red: 61 / 255.0, green: 152 / 255.0, blue: 60 / 255.0, alpha: 1)
        self.closeButton?.setGradientColor(from: Constants.appConfigColor.MessageButtonLeftColor, to: Constants.appConfigColor.MessageButtonRightColor, startPoint: CGPoint(x: -0.4, y: 0.5), toPoint: CGPoint(x: 1, y: 0.5))
        if let layer = self.closeButton?.layer.sublayers?.first {
            layer.cornerRadius = (self.closeButton?.frame.size.height ?? 2) / 2
            self.closeButton?.layer.cornerRadius = (self.closeButton?.frame.size.height ?? 2) / 2
        }

        self.messageLabel?.text = self.message
        self.view.autoresizingMask = []
        NotificationCenter.default.addObserver(self, selector: #selector(InvalidNetworkMessageViewController.appplicationDidEnterBackground(_:)), name: UIApplication.didEnterBackgroundNotification, object: nil)
    }

    override func viewWillLayoutSubviews() {
        super.viewWillLayoutSubviews()
        if !self.isRelayout {
            if let mnvc: MainNavViewController = self.presentingViewController as? MainNavViewController {
                if let cvc: InvalidNetworkRetryDelegate = mnvc.children.first as? InvalidNetworkRetryDelegate{
                    let topStart = cvc.controllerTopStart
                    self.view.frame = CGRect(x: 0, y: 0 - topStart, width: self.view.frame.size.width, height: self.view.frame.size.height + topStart)
                    self.isRelayout = true
                }
            }
        }
    }

    @IBAction func closeView(_ sender: NSObject) {
        self.dismiss(animated: true) {
            self.delegate?.refresh?()
        }
    }

    @objc func appplicationDidEnterBackground(_ notification: NSNotification) {
        self.dismiss(animated: true, completion: nil)
    }

    override func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)
        self.dismiss(animated: true, completion: nil)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent;
    }
}
